src/test/java/pages/HomePage.java

```diff
@@ -0,0 +1,31 @@
+package pages;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+public class HomePage extends BasePage {
+    public HomePage(WebDriver givenDriver) {
+        super(givenDriver);
+    }
+    @FindBy (css = ".playlist:nth-child(3)")
+    WebElement firstPlaylist;
+    @FindBy (css = "[name='name']")
+    WebElement playlistNameField;
+    @FindBy (css = "div.success.show")
+    WebElement renamePlaylistSuccessMsg;
+    public void doubleClickPlaylist() {
+        doubleClick((WebElement) firstPlaylist);
+    }
+    public void enterNewPlaylistName(String playlistName) {
+        findElement((WebElement) playlistNameField).sendKeys(Keys.chord(Keys.CONTROL,"a",Keys.BACK_SPACE));
+        findElement((WebElement) playlistNameField).sendKeys(playlistName);
+        findElement((WebElement) playlistNameField).sendKeys(Keys.ENTER);
+    }
+    public String getRenamePlaylistSuccessMsg() {
+        return findElement((WebElement) renamePlaylistSuccessMsg).getText();
```

Comment:
Hi,
Your homework 23 has been reviewed, sugestion: 
- Page Factory is an extension of the Page Object Model, which helps initialize WebElement objects more optimally and reduce code more concisely.
- It is used to initialize Page objects or to initialize the Page object itself. It is also used to initiate handler actions without using the driver.findElement / driver.findElements syntax.
=> So when you use `@FindBy` annotation for declaring Web Elements you don't need to add `driver.findElement`
// Example: 
```
public String getRenamePlaylistSuccessMsg() {
        return renamePlaylistSuccessMsg.getText();
 }
```
Please check other functions and edit them. (doubleClickPlaylist, enterNewPlaylistName)
💖Best wishes from the HW team (Khanh, khanh@testpro.io)!💖